docs/advanced/pool.rst

```diff
@@ -284,6 +284,45 @@ working connections, as soon as they are available.
 
 Faster than you can say poll. Or pool.
 
+Row factories
+-------------
+
+Pool's connections may be configured to use a :ref:`row factory
+<row-factories>`. One approach is to use the `!configure` constructor as
+follows:
+::
+
+    def set_row_factory(conn: psycopg.Connection[Any]) -> None:
+        conn.row_factory = psycopg.rows.class_row(MyRow)
+
+    pool = ConnectionPool[MyRow](configure=set_row_factory)
+    with pool.connection() as conn:
+        row = conn.execute(...).fetchone()
+        # row has type MyRow (or None)
+        # conn has type Connection[MyRow]
+
+Here it is needed to set ``ConnectionPool``'s type parameter to ``MyRow`` as
+the `!configure` callback is not enough for type inference.
+
+The same can be achieved by passing ``kwargs={"row_factory":
+psycopg.rows.class_row(MyRow)}`` to ``ConnectionPool``, but the type parameter
+for ``ConnectionPool`` remains needed.
```

Comment:
How about we also add a `row_factory` parameter to the pool constructor, and we assign the type parameter from there?

Reply:
We could make an error to specify row_factory both in the parameters and in the `kwargs` parameter.

Reply:
Indeed, this would be simpler. Thank you.